osg-comanage-project-usermap.py

```diff
@@ -146,6 +149,35 @@ def get_osguser_groups(filter_group_name=None):
              for pid, gids in pid_gids.items() }
 
 
+def parse_inputfile(inputfile):
+    user_groupmap = dict()
+    with open(inputfile) as file:
+        for line in file:
+            split_line = line.split()
+            if split_line[0] == "*" and len(split_line) >= 3:
+                user_groupmap[split_line[1]] = split_line[2].split(",")
+    return user_groupmap
+
+
+def merge_maps(maps):
+    while len(maps) > 1:
+        merge_target = maps[0]
+        merge_material = maps.pop(1)
+        for key in merge_material.keys():
```

Comment:
How about a for loop + new function-level variable to store the merged dict? That way you're not doing destructive ops on the input arg